lib/rules/jsx-tag-spacing.js

```diff
@@ -178,54 +180,68 @@ function validateAfterOpening(context, node, option) {
 
 function validateBeforeClosing(context, node, option) {
   // Don't enforce this rule for self closing tags
-  if (!node.selfClosing) {
-    const sourceCode = context.getSourceCode();
-    const lastTokens = sourceCode.getLastTokens(node, 2);
-    const closingToken = lastTokens[1];
-    const leftToken = lastTokens[0];
-
-    if (node.loc.start.line !== node.loc.end.line && option === 'multiline-always') {
-      if (leftToken.loc.end.line === closingToken.loc.start.line) {
-        report(context, messages.beforeCloseNeedNewline, 'beforeCloseNeedNewline', {
-          node,
-          loc: leftToken.loc.end,
-          fix(fixer) {
-            return fixer.insertTextBefore(closingToken, '\n');
-          },
-        });
-      }
-    }
-
-    if (leftToken.loc.start.line !== closingToken.loc.start.line) {
-      return;
-    }
+  if (node.selfClosing) {
+    return;
+  }
 
-    const adjacent = !sourceCode.isSpaceBetweenTokens(leftToken, closingToken);
+  const sourceCode = context.getSourceCode();
+  const leftToken = option === 'proportional-always'
+    ? getTokenBeforeClosingBracket(node)
+    : sourceCode.getLastTokens(node, 2)[0];
```

Comment:
does `getLastTokens` always return an array?

Reply:
Yes it does, at least that is what the eslint docs imply.